oss_scripts/pip_package/setup.py

```diff
@@ -73,8 +73,9 @@ def finalize_options(self):
     cmdclass={'install': InstallPlatlib},
     distclass=BinaryDistribution,
     install_requires=[
-        'tensorflow>=2.7.0rc0, <2.8',
-        'tensorflow_hub>=0.8.0',
+        "tensorflow>=2.7.0rc0, <2.8; platform_system != 'Darwin'",
+        "tensorflow-macos>=2.6.0, <2.7; platform_system == 'Darwin'",
```

Comment:
Darwin is all Macs yes? I believe you want darwin & arch is arm64 as the check.

Reply:
Yes, I think all Macs (including x86-64 and arm64) will install tensorflow-macos. 

Because tensorflow-macos relies on the [PluggableDevice interface](https://github.com/tensorflow/community/blob/master/rfcs/20200624-pluggable-device-for-tensorflow.md) to implement mac-enabled GPUs (including AMD and Apple M1). You can also see on this [page](https://pypi.org/project/tensorflow-macos/#files) that Apple does provide pre-compiled packages on x86-64 Macs. I think whether it is x86-64 or arm64 mac that does not use the standard version but uses the version built by apple, it is a better choice to get GPU acceleration.

 Finally, the details about tensorflow-macos can also be seen on the [Apple developer page](https://developer.apple.com/metal/tensorflow-plugin/).

Reply:
Apologies for the delayed reply, I've been busy getting some last items ready for the 2.7 release.

I see a few issues here:
1. tensorflow-macos is behind. There are features that we add to core TF that we require for our own releases. We would have to delay future mac releases which I do not want. Not to mention the tensorflow-macos does not provide any nightlies or release candidates.
2. I only see packages available for arm64 in tensorflow-macos: https://pypi.org/project/tensorflow-macos/#files
3. We build against tensorflow's macos targets. This means that the symbols in the .so files map to the shared lib of tensorflow. I would not expect any of our custom ops in tensorflow-text to work with tensorflow-macos without building specifically against it which would mean restructuring our entire build system.

Reply:
1. Yes, you are right, tensorflow-macos is behind, so I will replace the `x86-64 ` mac build with the standard TF.
2. Take a closer look, the second pre-compiled package is to support `x86-64`.
<img width="815" alt="截屏2021-11-09 下午2 10 45" src="https://user-images.githubusercontent.com/47380395/140873881-2fe6cdd2-1a62-4fc8-b7b5-a35b46e1a492.png">
3. I understand your worries and concerns, but `tensorflow-macos` itself does not have any additional custom operations, it just provides an interface for `tensorflow-metal` to use to call Mac's GPU (including AMD and Apple Silicon), use `tensorflow-macos` does not need to modify any source code, so it will not affect the development of TF.Text. If you don't use the plug-in `tensorflow-metal` ( [PluggableDevice interface](https://github.com/tensorflow/community/blob/master/rfcs/20200624-pluggable-device-for-tensorflow.md) does not affect the Core of TensorFlow), `tensorflow-macos` is just a pre-compiled package of tensorflow built by Apple, which is almost the same as the standard version.